tests/mesh/mesh_subdomain_id.C

```diff
@@ -0,0 +1,72 @@
+#include <libmesh/libmesh.h>
+#include <libmesh/elem.h>
+#include <libmesh/replicated_mesh.h>
+#include <libmesh/mesh_generation.h>
+
+#include "test_comm.h"
+#include "libmesh_cppunit.h"
+
+
+using namespace libMesh;
+
+class MeshSubdomainIDTest : public CppUnit::TestCase
+{
+  /**
+   * The goal of this test is to verify proper operation of the
+   * MeshBase::subdomain_ids() method.
+   */
+public:
+  CPPUNIT_TEST_SUITE( MeshSubdomainIDTest );
+
+  CPPUNIT_TEST( testUnpartitioned );
+  CPPUNIT_TEST( testMultiple );
+
+  CPPUNIT_TEST_SUITE_END();
+
+public:
+  void setUp() {}
+
+  void tearDown() {}
+
+  void testUnpartitioned()
+  {
+    std::unique_ptr<UnstructuredMesh> mesh = libmesh_make_unique<ReplicatedMesh>(*TestCommWorld);
```

Comment:
Why force ReplicatedMesh here?  If we use Mesh it'll test replicated or distributed depending on configure settings, and we'd expect it to work either way.